sympy/solvers/solvers.py

```diff
@@ -983,6 +984,30 @@ def _sympified_list(w):
             f[i] = fi
 
         if fi.is_Relational:
+            if flags.get('dict', False):
+                solution = reduce_inequalities(f, symbols=symbols)
+                if isinstance(solution, Equality):
+                    return [{solution.lhs: solution.rhs}]
```

Comment:
this looks bad, were are enumerating over things.  We probably shouldn't return one of them...

Certainly no criticism intended to author of this.  In particular I see other suspicious looking returns up above too!  Line 973 and 975 for example...